### PR TITLE
common/templates: add templates to pin and unpin messages

### PR DESCRIPTION
**Pull request by LemmeCry**
Being able to pin/unpin messages with the bot has been a common request. This PR does just that by adding custom command templates which allow pinning/unpinning a message.
The syntax for the template usage will be:
- `pinMessage channel messageID`
- `unpinMessage channel messageID`

### CODE REVIEW

**Comment by LemmeCry** [comment]
Those errors are handled automatically. Here is the error returned when you try to pin after the limit:
image

**Comment by Wolveric** [comment]
Yeah, that's the case I mean. Because it's difficult to track that count, I'd rather it not automatically panic because of that. I'd prefer the option to gracefully handle the error, by checking the return, indicating whether it failed.

File: common/templates/context_funcs.go [hunk 1]
```diff
@@ -402,6 +402,32 @@ func (c *Context) tmplEditMessage(filterSpecialMentions bool) func(channel inter
 	}
 }
 
+func (c *Context) tmplHandlePinMessage(mode bool) func(channel, msgID interface{}) (string, error) {
```
Comment (jo3-l) [inline common/templates/context_funcs.go:405]:
```suggestion
func (c *Context) tmplPinMessage(mode bool) func(channel, msgID interface{}) (string, error) {
```
For consistency with similar higher-order functions.

File: common/templates/context.go [hunk 2]
```diff
@@ -499,6 +499,8 @@ func baseContextFuncs(c *Context) {
 	c.addContextFunc("sendMessageNoEscapeRetID", c.tmplSendMessage(false, true))
 	c.addContextFunc("editMessage", c.tmplEditMessage(true))
 	c.addContextFunc("editMessageNoEscape", c.tmplEditMessage(false))
+	c.addContextFunc("pinMessage", c.tmplPinMessage(true))
```
Comment (jo3-l) [inline common/templates/context.go:502]: The values are 
```suggestion
	c.addContextFunc("pinMessage", c.tmplPinMessage(false))
```

The parameter specifies whether it should unpin.

File: common/templates/context.go [hunk 3]
```diff
@@ -499,6 +499,8 @@ func baseContextFuncs(c *Context) {
...
+	c.addContextFunc("unpinMessage", c.tmplPinMessage(false))
```
Comment (jo3-l) [inline common/templates/context.go:503]:
```suggestion
	c.addContextFunc("unpinMessage", c.tmplPinMessage(true))
```
Ditto.

**Review by jo3-l — CHANGES_REQUESTED** [review]

**Pull request merged**